tests/treemodel/node.js

```diff
@@ -178,4 +179,58 @@ describe( 'Node', () => {
 			} );
 		} );
 	} );
+
+	describe( 'toJSON', () => {
+		it( 'should serialize empty node', () => {
+			let node = new Element( 'one' );
+
+			expect( treeModelTestUtils.jsonParseStringify( node ) ).to.deep.equal( {
+				_attrs: [],
+				_children: {
+					_indexMap: [],
+					_nodes: []
+				},
+				name: 'one',
+				parent: null
+			} );
+		} );
+
+		it( 'should serialize node with attributes', () => {
+			let node = new Element( 'one', { foo: true, bar: false } );
+
+			expect( treeModelTestUtils.jsonParseStringify( node ) ).to.deep.equal( {
+				_attrs: [ [ 'foo', true ], [ 'bar', false ] ],
+				_children: {
+					_indexMap: [],
+					_nodes: []
+				},
+				name: 'one',
+				parent: null
+			} );
+		} );
+
+		it( 'should serialize node with children', () => {
+			expect( treeModelTestUtils.jsonParseStringify( root ) ).to.deep.equal( {
+				_attrs: [],
+				_children: {
+					_indexMap: [ 0, 1, 2 ],
```

Comment:
This is again question about private properties in serialised object. `_indexMap` is the private and internal mechanism to improve performance. If we change it none tests should fail. Now we would need to change these tests every time we change private class logics.